subworkflows/local/shortread_adapterremoval.nf

```diff
@@ -36,89 +31,51 @@ workflow SHORTREAD_ADAPTERREMOVAL {
 
     if ( params.shortread_clipmerge_mergepairs && !params.shortread_clipmerge_excludeunmerged ) {
 
-        ENSURE_FASTQ_EXTENSION1(
-            Channel.empty().mix(
+        ch_concat_fastq = Channel.empty()
+            .mix(
                 ADAPTERREMOVAL_PAIRED.out.collapsed,
                 ADAPTERREMOVAL_PAIRED.out.collapsed_truncated,
                 ADAPTERREMOVAL_PAIRED.out.singles_truncated,
-                ADAPTERREMOVAL_PAIRED.out.pair1_truncated,
-                ADAPTERREMOVAL_PAIRED.out.pair2_truncated
+                ADAPTERREMOVAL_PAIRED.out.paired_truncated
             )
-            .map { meta, reads ->
-                meta.single_end = true
-                [meta, reads]
-            }
-        )
-
-        CAT_FASTQ(
-            ENSURE_FASTQ_EXTENSION1.out.reads
-                .groupTuple()
-        )
+            .groupTuple()
+            .map { [it.head(), it.tail().flatten()] }
```

Comment:
```suggestion
            .map { [it.head(), it.tail().flatten()] }
```

I know this most likely works, but running `head` and `tail` is a bit fragile and isn't as clear. I would rather use it explicit with element IDs/indices because then someone else knows _Exactly_ what is being pulled (based on our data structure) - unless there is some performance reason why you went with that?

Reply:
No, that's a fair point.